source/module_base/global_file.cpp

```diff
@@ -153,28 +153,18 @@ void ModuleBase::Global_File::make_dir_out(
 #endif
     }
 
-    std::stringstream ss,ss1;
-
-    // mohan add 2010-09-12
-    if(out_alllog)
-    {
-	    ss << "running_" << calculation << "_" << rank + 1;
-	    open_log(GlobalV::ofs_running, ss.str(), calculation, restart);
-        #if defined(__CUDA) || defined(__ROCM)
-        open_log(GlobalV::ofs_device, "device" + std::to_string(rank), calculation, restart);
-        #endif
-    }
-    else
-    {
-	    if(rank==0)
-	    {
-		    ss << "running_" << calculation;
-		    open_log(GlobalV::ofs_running, ss.str(), calculation, restart);
-            #if defined(__CUDA) || defined(__ROCM)
-            open_log(GlobalV::ofs_device, "device", calculation, restart);
-            #endif
-	    }
-    }
+    auto open_running_log = [&](const std::string& keyword)
+        {
+            std::stringstream ss;
+            ss << "running_" << keyword;
+            if (out_alllog) { ss << "_" << rank + 1; }
+            open_log(GlobalV::ofs_running, ss.str(), keyword, restart);
+#if defined(__CUDA) || defined(__ROCM)
+            open_log(GlobalV::ofs_device, "device" + std::to_string(rank), calculation, restart);
+#endif
+        };
+    // if linear response solver only, open running_lr.log. else, open running_calculation.log
```

Comment:
what is the exact meaning of 'calculation?' why we need two names for lr? you write the code in a new way that makes lr so special, it's strange

Reply:
it's similar to what Peize wrote for the old exx code:   if(exx) {xxx} else {yyy}, here yyy is what most users are familiar with, such as scf and nscf, and 'xxx' is only a special case in my point of view. It's not good to write code in this special way to emphasize the exx code.